llama_index/vector_stores/types.py

```diff
@@ -65,23 +78,54 @@ class ExactMatchFilter(BaseModel):
     value: Union[StrictInt, StrictFloat, StrictStr]
 
 
+class AdvancedFilter(BaseModel):
+    """Advanced metadata filter for vector stores to support more operators.
+
+    Value uses Strict* types, as int, float and str are compatible types and were all
+    converted to string before.
+
+    See: https://docs.pydantic.dev/latest/usage/types/#strict-types
+    """
+
+    key: str
+    value: Union[StrictInt, StrictFloat, StrictStr]
+    operator: FilterOperator = FilterOperator.DEFAULT
+
+
 class MetadataFilters(BaseModel):
     """Metadata filters for vector stores.
 
     Currently only supports exact match filters.
     TODO: support more advanced expressions.
     """
 
+    # Exact match filters
     filters: List[ExactMatchFilter]
```

Comment:
sorry to be picky 😅 But I feel like adding an `advanced_filters` attribute is not the way to go? This is another kwarg that users have to learn.

Ideally, we can have something like 
`filters: List[Union[ExactMatchFilter, MetadataFilter]]` -- deprecate `ExactMatchFilter` and rename `AdvancedFilter` to `MetadataFilter` ?

We can add some backward compat function to `MetadataFilters` like `get_legacy_filters()`, which will only return exact match filters, so that other vector stores can use that and avoid breaking?

Reply:
that is my original intention, basically to deprecate `ExactMatchFilter` and have an unified `MetadataFilter `.
but it seems break a lot of things.
I will try to do this way and can you help for this part for just changing this pr. I will do the rest first

```
We can add some backward compat function to MetadataFilters like get_legacy_filters(), which will only return exact match filters, so that other vector stores can use that and avoid breaking?

```

Reply:
yea I can add the backwards compat, one sec